engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -543,8 +543,9 @@ public URLClassLoader loadDependencies()
 
   private void setTokenRefreshCredentials(LogicalPlan dag, Configuration conf) throws IOException
   {
-    String principal = StramUserLogin.getPrincipal();
-    String keytabPath = conf.get(StramClientUtils.KEY_TAB_FILE);
+    String principal = conf.get(StramClientUtils.TOKEN_REFRESH_PRINCIPAL, StramUserLogin.getPrincipal());
+    String keytabPath = principal.equals(StramUserLogin.getPrincipal()) ? conf.get(StramClientUtils.KEY_TAB_FILE, null) : conf.get(StramClientUtils.TOKEN_REFRESH_KEYTAB, null);
```

Comment:
If the user provided keytab is null then we should fall back to using the StramUserLogin's keytab.

This would be the System user's keytab.

I believe its necessary for Apex Cli based launch when the dt.authentication.store.keytab is not configured then it should be picked up from the local unde $HOME/.dt/dt-site.xml.

This is the current behavior and should remain consistent.